### ISSUE TITLE
Jade 'block unbuffered code' syntax fails—update Jade?

### ISSUE THREAD

**Issue by lunelson** [sintaxi/terraform#114]
The block unbuffered syntax for writing JS in Jade, [described here](http://jade-lang.com/reference/code/) does not seem to work—could it be that jade is not up to date?

```jade
-
  list = ["Uno", "Dos", "Tres",
          "Cuatro", "Cinco", "Seis"]
each item in list
  li= item
```

**Comment by chharvey** [comment]
according to #115 the jade dependency has been upgraded to **jade v1.11.0**, which supports block unbuffered code, however I am still getting errors with such code in my projects. I am using harp **v0.20.1**. any news on this?

**Comment by lunelson** [comment]
This may have been merged in terraform but I'm not sure the terraform bump is merged in harp yet. You'd have to look at the current release of harp and check which terraform release the package is pointing to.

OTOH if it's in and you're still having trouble it could be because the syntax is quite strict. You have to put a single `-` followed by a `\n`, and then indent your block, and make sure something is written on the first line of the indented block. If there's a space or anything else after your `-` character it won't work, and if the first line of your code block is empty it will also fail

```jade
-
  // js code will work here only if previous line has no space after the '-' character
  // ... and only if there's no empty line after the '-' character
  function hello() { return 'hello world'; }
```

**Comment by chharvey** [comment]
The syntax is exactly correct; it's not an issue with that.

It looks like #115 has been merged into master (see https://github.com/sintaxi/terraform/commit/b7072dd767759ef02787fa7067f56a9dad052af1) before **v1.0.0** so no work needs to be done on terraform. But harp still needs to upgrade to **terraform v1.0.0** which is covered here: https://github.com/sintaxi/harp/pull/526

**Comment by lunelson** [comment]
Yeah that's what I figured, I just mentioned the possible syntax issue cause it's bitten me a few times. That PR re upgrading harp to terraform v1.0.0 has been in the queue for a while. I have the impression the maintainers are intending a major release of harp that will bundle a bunch of new features at once. In the meantime I've been using my forked version of harp with good success for a while. It merges some of the recent PRs must mainly just points to my fork of terraform, where I've merged a lot of the PRs that are currently in the queue over there, as well as my own features. I just `npm link` them on my own systems so I can tweak them both as I go.
